### DISCUSSION TITLE
Uploading files/images to google cloud storage?

### DISCUSSION THREAD

**Discussion by kyle-hamlin** [fastapi/fastapi#8255] [Questions] [answered]
If there a recommended way of uploading files and images that are database fields to cloud storage providers such as Google/S3? Perhaps something similar to how Django does it? https://docs.djangoproject.com/en/2.1/ref/files/storage/

**Comment by tiangolo** [comment, marked as answer]
You could use any library you want, there's nothing strongly required/integrated. For example, you could directly use [`apache-libcloud`](https://libcloud.apache.org/) (I'm currently using it for a project), or [`minio`](https://github.com/minio/minio-py), or even [`boto3`](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html).

**Comment by tiangolo** [comment]
I'll assume you were able to solve your use case, so I'll close this issue now, but feel free to add more comments or create new issues.

**Comment by techxonia** [comment]
@tiangolo Can you please share some demo project for uploading file to cloud storage using apache-libcloud?

**Comment by tiangolo** [comment]
Sorry @techxonia, I'm not using that library.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
